Sources/swift-bootstrap/main.swift

```diff
@@ -108,6 +108,16 @@ struct SwiftBootstrapBuildTool: ParsableCommand {
     @Flag()
     public var useIntegratedSwiftDriver: Bool = false
 
+    /// A flag that indicates this build should check whether targets only import
+    /// their explicitly-declared dependencies
+    @Option()
+    public var explicitTargetDependencyImportCheck: TargetDependencyImportCheckingMode = .none
+
+    enum TargetDependencyImportCheckingMode: String, Codable, ExpressibleByArgument {
+        case none
+        case error
```

Comment:
I explicitly left out `warn` here in the spirit of keeping `swift-bootstrap` minimal.